test/index-test.js

```diff
@@ -6,13 +6,152 @@ const path = require('path')
+        var coolSpy = expect.spyOn(cool, "call")
+        bake.call(pie, update)
+        setTimeout(function() {
+          expect(coolSpy).toHaveBeenCalledWith(pie, update)
+          coolSpy.restore()
+          done()
+        }, 3000)
+      })
+
+    })
+
+    describe('mix', () => {
+      it('calls the update function', () => {
+        mix.call(pie, update)
+        expect(document.getElementById("pie").getElementsByClassName("status")[0].innerText).toMatch(/Mixing/)
+      })
+
+      it('calls bake with context', function(done) {
+        this.timeout(5000)
+        var bakeSpy = expect.spyOn(bake, "call")
+        mix.call(pie, update)
+        setTimeout(function() {
+          expect(bakeSpy).toHaveBeenCalledWith(pie, update)
+          bakeSpy.restore()
+          done()
+        }, 3000)
+      })
+
+    })
   })
+
+  describe('makeDessert', () => {
+    it('calls the right function', () => {
+      var pieNode = document.getElementById("pie")
+      var cakeNode = document.getElementById("cake")
+      var makeCakeSpy = expect.spyOn(makeCake, "call")
+      var makePieSpy = expect.spyOn(makePie, "call")
+      makeDessert.call(pieNode.getElementsByClassName("js-make")[0])
+      makeDessert.call(cakeNode.getElementsByClassName("js-make")[0])
+      expect(makePieSpy).toHaveBeenCalledWith(pieNode)
+      expect(makeCakeSpy).toHaveBeenCalledWith(cakeNode)
+      makeCakeSpy.restore()
+      makePieSpy.restore()
+    })
+  })
+
+  describe('cake', () => {
+    it('has a decorate function', () => {
+      expect(cake.decorate).toExist()
+    })
+  })
+
+  describe('pie', () => {
+    it('borrows the decorate function from cake', () => {
+      var spy = expect.spyOn(cake.decorate, "bind").andCallThrough()
+      makePie.call(document.getElementById("pie"))
+      expect(pie.decorate).toExist()
+      expect(spy).toHaveBeenCalledWith(pie)
+    })
+  })
+
+  describe('makePie', () => {
+    it('binds an update function', () => {
+      var spy = expect.spyOn(updateStatus, "bind").andCallThrough()
+      var node = document.getElementById("pie")
+      makePie.call(node)
+      expect(spy).toHaveBeenCalledWith(node)
+      spy.restore()
+    })
+
+    it('calls mix with context', () => {
+      var spy = expect.spyOn(mix, "call").andCallThrough()
+      var node = document.getElementById("pie")
+      makePie.call(node)
+      expect(spy.calls[0].arguments[0]).toBe(pie)
+      spy.restore()
+    })
+  })
+
+  describe('makeCake', () => {
+    it('binds an update function', () => {
+      var spy = expect.spyOn(updateStatus, "bind").andCallThrough()
+      var node = document.getElementById("cake")
+      makeCake.call(node)
+      expect(spy).toHaveBeenCalledWith(node)
+      spy.restore()
+    })
+    it('calls mix with context', () => {
+      var spy = expect.spyOn(mix, "call").andCallThrough()
+      var node = document.getElementById("cake")
+      makeCake.call(node)
+      expect(spy.calls[0].arguments[0]).toBe(cake)
```

Comment:
`toHaveBeenCalledWith` needs all args and I was only testing for the one, but easy enough to bind an update function and check against it, probably a better test.